### PR TITLE
Add pulseaudio server for audio.

### PR DESCRIPTION
**Pull request by padmashree9110**
This patch adds pulseaudio server path for
audiobackend driver pa.

Signed-off-by: pmandri <padmashree.mandri@intel.com>

### CODE REVIEW

File: src/guest/start.c [hunk 1]
```diff
@@ -45,7 +45,7 @@ static const char *fixed_cmd =
 	" -device usb-mouse"
 	" -device usb-kbd"
 	" -device intel-hda -device hda-duplex,audiodev=android_spk"
-	" -audiodev id=android_spk,timer-period=5000,driver=pa,in.fixed-settings=off,out.fixed-settings=off"
+	" -audiodev id=android_spk,timer-period=5000,driver=pa,server=/run/user/1000/pulse/native,in.fixed-settings=off,out.fixed-settings=off"
```
Comment (YadongQi) [inline src/guest/start.c:48]: Do not hardcode 1000 here, the user id might not 1000

**Review by YadongQi — CHANGES_REQUESTED** [review]

File: src/guest/start.c [hunk 2]
```diff
@@ -44,12 +44,22 @@ static const char *fixed_cmd =
 	" -device qemu-xhci,id=xhci,p2=8,p3=8"
 	" -device usb-mouse"
 	" -device usb-kbd"
-	" -device intel-hda -device hda-duplex,audiodev=android_spk"
-	" -audiodev id=android_spk,timer-period=5000,driver=pa,in.fixed-settings=off,out.fixed-settings=off"
 	" -device e1000,netdev=net0"
 	" -device intel-iommu,device-iotlb=on,caching-mode=on"
 	" -nodefaults ";
 
+int get_uid() {
```
Comment (YadongQi) [inline src/guest/start.c:51]: make it static if it is only used in this file.

File: src/guest/start.c [hunk 3]
```diff
@@ -1141,9 +1151,11 @@ int start_guest(char *name)
 		}
 		g_strfreev(srv_list);
 	}
-
 	cx = snprintf(p, size, "%s", fixed_cmd);
 	p += cx; size -= cx;
+	uid= get_uid();
+	cx = snprintf(p, size,"-device intel-hda -device hda-duplex,audiodev=android_spk -audiodev id=android_spk,timer-period=5000,driver=pa,server=/run/user/%d/pulse/native,in.fixed-settings=off,out.fixed-settings=off", uid);
+	p += cx; size -= cx;
```
Comment (YadongQi) [inline src/guest/start.c:1158]: move this block code to above extra groups (line 1129) since it is not fixed cmd any more.

**Review by YadongQi — CHANGES_REQUESTED** [review]

**Review by YadongQi — APPROVED** [review]

**Review by gkdeepa — APPROVED** [review]
looks good to me

**Pull request merged**